addons/html_editor/static/src/core/history_plugin.js

```diff
@@ -366,6 +361,18 @@ export class HistoryPlugin extends Plugin {
     }
     stageSelection() {
         const selection = this.shared.getEditableSelection();
+        // TODO: Should not have an attribute mutation too.
+        // When you apply a mutation, you should add it in a step.
```

Comment:
Would be nice to have a comment that explains why we stage the selection and when it is problematic to stage the selection, something like:
- whenever we want to revert a step, we revert all the mutations, then set the selection to where it was just before any mutation
- It means that until we want to apply any mutation in the dom, we should keep track of the latest selection
- As soon as we make a mutation we cannot longer update the current step selection, because if the selection is impossible whenever revert all the mutation, a crash will happen